todo_app/data/todo_items.py

```diff
@@ -0,0 +1,153 @@
+        self.id = id
+        self.name = name
+        self.idList = idList
+        self.due_date = due_date
+        self.description = description
+        self.modified = modified
+    
+    def get_date(self):
+        return datetime.datetime.strftime(self.due_date, '%Y-%m-%d')
+    
+    def get_user_date(self):
+        return datetime.datetime.strftime(self.due_date, '%d/%m/%Y')
+
+    def get_modified_date(self):
+        return datetime.datetime.strftime(self.modified, '%Y-%m-%d')
+    
+    def get_modified_user_date(self):
+        return datetime.datetime.strftime(self.modified, '%d/%m/%Y')
+    
+    def get_card_as_dictionary(self):
+        return {'name' : self.name, 'idList' : self.idList, 'due_date' : self.due_date, 'description' : self.description, 'modified' : self.modified}
+
+class ViewModel:
+    def __init__(self, items, lists):
+        self._items = items
+        self._lists = lists
+
+    @property
+    def items(self):
+        return self._items
+
+    @property
+    def lists(self):
+        return self._lists
+
+    @property
+    def items_todo(self):
+        items = []
+        for item in self._items:
+            if item.idList == self.lists['ToDo']:
+                items.append(item)
+        return items
+
+    @property
+    def items_doing(self):
+        items = []
+        for item in self._items:
+            if item.idList == self.lists['Doing']:
+                items.append(item)
+        return items
+
+    @property
+    def items_done(self):
+        items = []
+        for item in self._items:
+            if item.idList == self.lists['Done']:
+                items.append(item)
+        return items
+
+    @property
+    def recent_done_items(self):
+        items = []
+        for item in self._items:
+            if item.idList == self.lists['Done'] and item.modified == datetime.date.today():
+                items.append(item)
+        return items
+
+    @property
+    def older_done_items(self):
+        items = []
+        for item in self._items:
+            if item.idList == self.lists['Done'] and item.modified != datetime.date.today():
+                items.append(item)
+        return items
+
+def get_mongodb_connection():
+    mongodb_url = os.getenv('MONGO_URL')
+    return mongodb_url
+
+def get_mongodb_database_name():
+    mongodb_name = os.getenv('MONGO_DB_NAME')
+    return mongodb_name
+
+def mongo_db_connection():
+    db_connection = get_mongodb_connection()
+    db_name = get_mongodb_database_name()
+
+    mongo_client = pymongo.MongoClient(db_connection)
+    db = mongo_client[db_name]
+
+    return db
+
+def get_todo_cards():
+    db = mongo_db_connection()
+    collections = db.list_collection_names()
+
+    card_list = []
+    for col in collections:
+        collection = db[col]
+
+        for card in collection.find({}):
+            card_list.append(ToDoCard(card['_id'], card['name'], card['idList'], card['due_date'], card['description'], card['modified']))
```

Comment:
You currently have idList on the card which is effectively duplicating information - the card is in a collection called e.g. 'todo', and then also has idList = 'todo'.

Pick one or the other. Have a single collection where each card has an 'idList' to store its current status, **or** have three collections and the card has status 'todo' if it is in the collection called 'todo'. To do the latter here, you would delete any mention of idLuse `col` in place of `card['idList']`.

The risk with duplicating information is added complexity, and what will you do if there's a mistake and you end up with a card with an idList different from its collection - which is true?